### PR TITLE
[backend/frontend] Add participants to cases and other entities(#3180)

### PR DESCRIPTION
**Pull request by yassine-ouaamou**
This is the final PR for the issue 3180.
The reviews that were made on the [first one](https://github.com/OpenCTI-Platform/opencti/pull/3614) have been taken into account.

### Proposed changes

* Add Participants to Incident Response, Request for Information, Request for Takedown, Tasks, Reports and Incidents
* Add filters on Participants on the lists of these entities
* Add a Participant filter on dashboards
* Replace the column Author by Assignees in Incident Response, Request for Information and Request for Takedown

### Related issues

* #3180 
*

### Checklist

- [ ] I consider the submitted work as finished
- [ ] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/components/ItemParticipants.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { FunctionComponent } from 'react';
+import Button from '@mui/material/Button';
+
+interface ItemParticipantsProps {
+  // TODO To fix tomorrow :)
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/components/ItemParticipants.tsx]: never put off until tomorrow what you can do today 💐

File: opencti-platform/opencti-front/src/components/ItemParticipants.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  participantsEdges,
+}) => {
+  const participants = participantsEdges.slice()
+    .sort((a, b) => a.node.name.localeCompare(b.node.name))
+    .map((n) => n.node);
+
+  return (
+    <div>
+      {participants.length > 0
+        ? participants.map((assignee) => (
```
Comment (marieflorescontact) [inline opencti-platform/opencti-front/src/components/ItemParticipants.tsx]: Should't it be `participants.map((participant)` ?

File: opencti-platform/opencti-front/src/components/ItemParticipants.tsx (hunk 1 above)
Comment (yassine-ouaamou) [reply opencti-platform/opencti-front/src/components/ItemParticipants.tsx]: It was fixed (today :D)

File: opencti-platform/opencti-front/src/components/ItemParticipants.tsx (hunk 2 above)
Comment (yassine-ouaamou) [reply opencti-platform/opencti-front/src/components/ItemParticipants.tsx]: you are right!

**Review by RomuDeuxfois — APPROVED** [review]

**Comment by helene-nguyen** [comment]
- Can select the particpants and assignees twice

image

**Comment by helene-nguyen** [comment]
Locally tested and looks good.

**Comment by RomuDeuxfois** [comment]
> * Can select the particpants and assignees twice
> 
> image

It's a global behavior, we need to find a solution for all autocomplete fiels (participants, assignees, etc)

**Pull request merged**